providers/yttcb/pinniped_addon_secret.lib.yaml

```diff
@@ -0,0 +1,204 @@
+      bindDN: #@ data.values.LDAP_BIND_DN
+      BIND_PW_ENV_VAR: #@ data.values.LDAP_BIND_PASSWORD
+      usernamePrompt: LDAP Username
+      insecureSkipVerify: false
+      userSearch:
+        baseDN: #@ data.values.LDAP_USER_SEARCH_BASE_DN
+        filter: #@ data.values.LDAP_USER_SEARCH_FILTER
+        username: #@ data.values.LDAP_USER_SEARCH_USERNAME
+        idAttr: #@ data.values.LDAP_USER_SEARCH_ID_ATTRIBUTE
+        emailAttr: #@ data.values.LDAP_USER_SEARCH_EMAIL_ATTRIBUTE
+        nameAttr: #@ data.values.LDAP_USER_SEARCH_NAME_ATTRIBUTE
+        scope: sub
+      groupSearch:
+        baseDN: #@ data.values.LDAP_GROUP_SEARCH_BASE_DN
+        filter: #@ data.values.LDAP_GROUP_SEARCH_FILTER
+        nameAttr: #@ data.values.LDAP_GROUP_SEARCH_NAME_ATTRIBUTE
+        scope: sub
+        userMatchers: #! Add more matchers if necessary
+          - userAttr: #@ data.values.LDAP_GROUP_SEARCH_USER_ATTRIBUTE
+            groupAttr: #@ data.values.LDAP_GROUP_SEARCH_GROUP_ATTRIBUTE
+    oauth2:
+      skipApprovalScreen: true
+      responseTypes:
+        - "code"
+    storage:
+      type: kubernetes
+      config:
+        inCluster: true
+    enablePasswordDB: false
+  service: #@ getServiceValuesForMC(dex_service_name)
+  deployment:
+    replicas: 1
+  certificate: #! required
+    duration: #@ data.values.CERT_DURATION
+    renewBefore: #@ data.values.CERT_RENEW_BEFORE
+  dns: #! required
+    vsphere:
+      dnsNames:
+        - tkg-dex.com
+      ipAddresses: [ 0.0.0.0 ] #! Do not change. Will be updated by post-deployment job
+    aws:
+      dnsNames:
+        - tkg-dex.com
+      DEX_SVC_LB_HOSTNAME: example.com #! <DEX_SVC_LB_HOSTNAME> is required for AWS
+    azure:
+      dnsNames:
+        - tkg-dex.com
+      DEX_SVC_LB_HOSTNAME: dex.example.com #! <DEX_SVC_LB_HOSTNAME> is required for azure
+#@ end
+
+#@ def getValuesForWC():
+#@ bomData = get_bom_data_for_tkr_name()
+#@ pinnipedImage = bomData.components["pinniped"][0].images.pinnipedImage
+#@ postDeployPinnipedImage = bomData.components["pinniped-post-deploy"][0].images.tkgPinnipedPostDeployImage
+---
+infrastructure_provider: #@ data.values.PROVIDER_TYPE
+tkg_cluster_role: #@ data.values.TKG_CLUSTER_ROLE
+identity_management_type: #@ data.values.IDENTITY_MANAGEMENT_TYPE
+pinniped:
+  cert_duration: #@ data.values.CERT_DURATION
+  cert_renew_before: #@ data.values.CERT_RENEW_BEFORE
+  supervisor_svc_endpoint: #@ data.values.SUPERVISOR_ISSUER_URL
+  supervisor_ca_bundle_data: #@ data.values.SUPERVISOR_ISSUER_CA_BUNDLE_DATA_B64
+  image: #! For backwards compatibility. i.e. C.1 mgmt cluster creates a C.0 wlc with older addon templates.
+    name: #@ pinnipedImage.imagePath
+    tag: #@ pinnipedImage.tag
+    repository: #@ get_image_repo_for_component(pinnipedImage)
+    pull_policy: IfNotPresent
+  post_deploy_job_image: #! For backwards compatibility. i.e. C.1 mgmt cluster creates a C.0 wlc with older addon templates.
```

Comment:
@shyaamsn also asked if we can remove `post_deploy_job_image`.  Calgary is quite old.

Reply:
Removed.